### PR TITLE
Countries acceptance feature tests

### PR DESCRIPTION
**Pull request by krivenkop**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -11,4 +11,8 @@ class User < ApplicationRecord
   def full_name
     first_name + ' ' + last_name
```
Comment (Codcore) [inline app/models/user.rb]: Let use interpolation:
```
"#{first_name} #{last_name}"

File: spec/factories/admins.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+FactoryBot.define do
+  factory :admin do
+    email { "admin@gmail.com" }
```
Comment (Codcore) [inline spec/factories/admins.rb]: Let use `user` factory with `admin` trait(take a look at official docs):
```
let(:admin) { create :user, admin: true }
```

File: spec/factories/admins.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    password { "12345678" }
+    first_name { "Steve" }
+    last_name { "Voznyak" }
```
Comment (Codcore) [inline spec/factories/admins.rb]: You can use sequences for `first_name` and `last_name` for example

File: spec/features/countries/admin.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+} do
+  context "User is admin" do
+    let(:admin) { create(:admin) }
+
+    before { sign_in admin }
+
+    context "Admin see an empty list if no items in db" do
+      get admin_countries_path
+
+      expect(page).to have_content 'No items'
```
Comment (Codcore) [inline spec/features/countries/admin.rb]: It is good covention to limit scope of `have_content` matcher:l, for example:
```
within ".div.container" do
  expect(page).to have_content 'No items'
end
```

File: spec/features/countries/admin.rb [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    end
+
+    context "Admin can see a country from db" do
+      country = Country.new(attributes_for :country)
```
Comment (Codcore) [inline spec/features/countries/admin.rb]: Instead use `FactoryBot::build`
https://devhints.io/factory_bot

File: spec/features/countries/admin.rb [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+      expect(page).to have_content country['name']
+      expect(page).to have_content country['alpha_three_code']
+    end
+
+    context "Admin adds new no valid country" do
+      get admin_countries_path
+      expect(page).to have_content 'No items'
+
+      click_on 'Add new'
```
Comment (Codcore) [inline spec/features/countries/admin.rb]: Use scoping

File: spec/features/countries/admin.rb [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+      country_attrs = attributes_for :country
+      fill_in 'Name', country_attrs['name']
+      click_on 'Add'
```
Comment (Codcore) [inline spec/features/countries/admin.rb]: Same

File: spec/features/countries/admin.rb [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+      expect(page).to have_content 'Alpha three code can not be empty'
+    end
+
+
+    context "Admin removes country" do
+      country = Country.new(attributes_for :country)
```
Comment (Codcore) [inline spec/features/countries/admin.rb]: Use `FactoryBot::build`

File: spec/features/countries/admin.rb [hunk 9]
```diff
@@ -0,0 +1,92 @@
+require 'rails_helper'
+
```
Comment (Codcore) [inline spec/features/countries/admin.rb]: file should have name `spec/features/countries/admin_spec.rb`

File: spec/features/countries/admin.rb [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+feature "Add new country via admin panel", %q{
+  For access to creating, seeing, updating
+  and deleting of countries
+  user need to login on website and be admin
+} do
+  context "User is admin" do
+    let(:admin) { create(:admin) }
```
Comment (Codcore) [inline spec/features/countries/admin.rb]: In feature specs use `given` instead `let`, it is overall convention.

File: spec/features/countries/admin.rb [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+    before { sign_in admin }
```
Comment (Codcore) [inline spec/features/countries/admin.rb]: In feature specs use `background` instead `before`, it is overall convention.

File: spec/features/countries/admin.rb [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+      click_on '.country__delete'
+      click_on 'Confirm'
+
+      expect(page).to have_content 'No items'
+    end
+  end
+
+  context "User is authenticated and not admin" do
+    let(:user) { create(:user) }
+    before { sign_in user }
```
Comment (Codcore) [inline spec/features/countries/admin.rb]: We already have that code, let move it to the root before top `context`

Comment (krivenkop) [reply spec/features/countries/admin.rb]: We have different types of user in another contexts

File: spec/factories/users.rb [hunk 13]
```diff
@@ -1,16 +1,13 @@
 FactoryBot.define do
   factory :user do
-    email { "test@gmail.com" }
-    password { "12345678" }
-    first_name { "Alex" }
-    last_name { "Jobs" }
-    username { "alex_jobs" }
-  end
+    email { 'test@gmail.com' }
```
Comment (Codcore) [inline spec/factories/users.rb]: Use sequences for fields that should be uniq:
```
sequence :email do |n|
    "person#{n}@example.com"
  end
```

**Pull request merged**